server/src/main/java/io/crate/action/sql/Session.java

```diff
@@ -170,6 +171,7 @@ public void quickExec(String statement, ResultReceiver<?> resultReceiver, Row pa
      */
     public void quickExec(String statement, Function<String, Statement> parse, ResultReceiver<?> resultReceiver, Row params) {
         CoordinatorTxnCtx txnCtx = new CoordinatorTxnCtx(sessionContext);
+        NodeContext nodeCtx = new NodeContext(planner.functions());
```

Comment:
I think the `NodeContext` could be created eagerly in the constructor - it should be valid over the life-time of a node.


We could even consider making it a (injected?) singleton.
This likely affects a lot of the other choices in the diff as well.

Reply:
In the `MetadataModule` class we bind functions/schemas as eager singletons, can I make NodeContext encapsulate both?